src/components/ActionIconInput.tsx/ActionIconInput.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  ActionIcon,
+  ActionIconProps,
+  Group,
+  Popover,
+  PopoverProps,
+  TextInput,
+} from '@mantine/core';
+import { getHotkeyHandler, useClickOutside } from '@mantine/hooks';
+import { useState } from 'react';
+
+export function ActionIconInput<T>({
+  onSubmit,
+  actionIconProps,
+  children,
+  placeholder,
+  ...popoverProps
+}: {
+  onSubmit: (input: string) => void;
+  children: React.ReactNode;
+  placeholder?: string;
+  actionIconProps?: ActionIconProps;
+} & Omit<PopoverProps, 'children'>) {
+  const [editing, setEditing] = useState(false);
+  const [control, setControl] = useState<HTMLInputElement | null>(null);
+  const [toggle, setToggle] = useState<HTMLButtonElement | null>(null);
```

Comment:
Weird that we are using state instead of ref to handle input changes, but I think I understand why